src/System.Private.Uri/src/System/PooledCharArray.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+
+namespace System
+{
+    internal struct PooledCharArray
+    {
+    	private char[] _buffer;
+
+    	public PooledCharArray(int size)
+    	{
+    		_buffer = ArrayPool<char>.Shared.Rent(size);
+    	}
+
+		public PooledCharArray(char[] fromArray)
+		{
+			_buffer = ArrayPool<char>.Shared.Rent(fromArray.Length);
+			for (int i = 0; i < fromArray.Length; i++)
+			{
+				_buffer[i] = fromArray[i];
+			}
+		}
+
+    	public void Release()
+    	{
+    		if (_buffer != null)
+    		{
+    			ArrayPool<char>.Shared.Return(_buffer);
+    			_buffer = null;
+    		}
+    	}
+
+    	public int Length => _buffer.Length;
+
+		public char[] Array => _buffer;
+
+    	public string GetStringAndRelease(int stringLength)
+    	{
+    		string ret = new string(_buffer, 0, stringLength);
+    		Release();
+    		return ret;
+    	}
+
+		public void CopyAndRelease(char[] dest, int offset, int count)
```

Comment:
>offset [](start = 46, length = 6)

you don't use offset so it might just make sense to have a single parameter (length)